samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -841,8 +842,11 @@ class SamzaContainer(
   def startLocalityManager {
     if(localityManager != null) {
       info("Registering localityManager for the container")
-      localityManager.start
-      localityManager.register(String.valueOf(containerContext.id))
+      // Ideally coordinator stream manager should be directly, but the SamzaContainer constructor parameter list is
```

Comment:
Should be ok to add 1 more param, keeps the dependencies clearer. The list too long is a symptom of a different problem. This way you can also remove the public getCoordinatorStreamManager() from LocalityManager.